ResearchKit/Common/ORKLocationSelectionView.m

```diff
@@ -0,0 +1,343 @@
+#import <MapKit/MapKit.h>
+#import <AddressBookUI/AddressBookUI.h>
+#import "ORKAnswerTextField.h"
+#import "ORKHelpers.h"
+#import "ORKAnswerFormat_Internal.h"
+
+@interface ORKLocationSelectionView () <UITextFieldDelegate, MKMapViewDelegate, CLLocationManagerDelegate>
+
+@property (nonatomic, strong) NSLayoutConstraint *mapViewHeightConstraint;
+@property (nonatomic, strong) NSLayoutConstraint *textFieldBottomConstraint;
+@property (nonatomic, strong) ORKAnswerTextField *textField;
+@end
+
+@implementation ORKLocationSelectionView {
+    CLLocationManager *_locationManager;
+    MKPointAnnotation *_selectedLocationAnnotation;
+    UIButton *_currentLocationButton;
+    BOOL _userLocationNeedsUpdate;
+    MKCoordinateRegion _answerRegion;
+}
+
+- (instancetype)init {
+    
+    self = [super init];
+    
+    if (self) {
+        
+        _textField = [[ORKAnswerTextField alloc] init];
+        _textField.delegate = self;
+        _textField.placeholder = ORKLocalizedString(@"LOCATION_ADDRESS",nil);
+        _textField.clearButtonMode = UITextFieldViewModeWhileEditing;
+        _textField.returnKeyType = UIReturnKeySearch;
+        _textField.adjustsFontSizeToFitWidth = YES;
+        
+        [self addSubview:_textField];
+        
+        _currentLocationButton = [[UIButton alloc] init];
+        [_currentLocationButton setImage:[UIImage imageNamed:@"locationOff" inBundle:[NSBundle bundleForClass:[self class]] compatibleWithTraitCollection:nil] forState:UIControlStateNormal];
+        [_currentLocationButton setImage:[UIImage imageNamed:@"locationOn" inBundle:[NSBundle bundleForClass:[self class]] compatibleWithTraitCollection:nil] forState:UIControlStateHighlighted];
+        [_currentLocationButton addTarget:self action:@selector(loadCurrentLocation) forControlEvents:UIControlEventTouchUpInside];
+        [self addSubview:_currentLocationButton];
+        
+        NSDictionary *views = NSDictionaryOfVariableBindings(_textField, _currentLocationButton);
+        ORKEnableAutoLayoutForViews([views allValues]);
+        
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_textField(44.0)]" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:views]];
+        _textFieldBottomConstraint = [NSLayoutConstraint constraintWithItem:_textField attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0];
+        [self addConstraint:_textFieldBottomConstraint];
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-(20.0)-[_textField]-[_currentLocationButton(20.0)]-(20.0)-|" options:NSLayoutFormatDirectionLeadingToTrailing metrics:nil views:views]];
+        [self addConstraint:[NSLayoutConstraint constraintWithItem:_currentLocationButton attribute:NSLayoutAttributeCenterY relatedBy:NSLayoutRelationEqual toItem:_textField attribute:NSLayoutAttributeCenterY multiplier:1 constant:0]];
```

Comment:
Move constraint code into method `setUpConstraints`. And use `activateConstraints`.